src/Workleap.DomainEventPropagation.Analyzers/EventDomainAttributeUsageAnalyzer.cs

```diff
@@ -61,13 +72,27 @@ public void AnalyzeOperationInvocation(SymbolAnalysisContext context)
             {
                 if (this.ImplementsBaseDomainEventInterface(classTypeSymbol))
                 {
-                    var hasDomainEventAttribute = classTypeSymbol.GetAttributes()
-                        .Any(x => x.AttributeClass != null && SymbolEqualityComparer.Default.Equals(x.AttributeClass, this._domainEventAttributeType));
+                    var domainEventAttribute = classTypeSymbol.GetAttributes()
+                        .FirstOrDefault(x => x.AttributeClass != null && SymbolEqualityComparer.Default.Equals(x.AttributeClass, this._domainEventAttributeType));
 
-                    if (!hasDomainEventAttribute)
+                    if (domainEventAttribute is null)
                     {
                         context.ReportDiagnostic(UseDomainEventAttribute, classTypeSymbol);
                     }
+
+                    if (domainEventAttribute is not null && domainEventAttribute.ConstructorArguments.Length == 1)
+                    {
+                        var attributeArgument = domainEventAttribute.ConstructorArguments[0].Value;
+                        if (attributeArgument is string attributeArgumentString)
+                        {
+                            var wasAdded = this._existingAttributes.TryAdd(attributeArgumentString, string.Empty);
```

Comment:
A nice improvement would be to disclose the name of the other symbol that have the same attribute ctor argument value. But I'm fine for this implementation for a V1.